### ISSUE TITLE
Make filter dropdown code generalized

### ISSUE THREAD

**Issue by crutchcorn** [unicorn-utterances/unicorn-utterances#2]
While work on this was started in terms of making the a11y code hooks based, it was never finished.

I should be able to have a dropdown button that will resize based on the "button"'s text size/etc without having to have it be marked as an aria-listbox

**Comment by crutchcorn** [comment]
Working on this now
